### PR TITLE
Refactor data sources and add data table in Poke

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR introduces a new reusable component called `DataTable`, which enables the display of data in a tabular format with row actions. The component will be used to refactor the workspace view to show data sources in the `DataTable`. This PR also includes the necessary files from the [shadcn](https://ui.shadcn.com/docs/components/data-table) library to use the DataTable component, but it will not be exposed outside of Poke (seen with @Duncid). Follow-up PRs will be submitted to refactor the rest of the UI using the new `DataTable` component.

workspace-data-table-data-source

### CODE REVIEW

File: front/components/poke/shadcn/ui/data_table.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
+"use client";
```
Comment (fontanierh) [inline front/components/poke/shadcn/ui/data_table.tsx:1]: I don't think we should have this ? This only makes sense if we have RSC

**Review by lasryaric — APPROVED** [review]
LGTM.

**Review by fontanierh — APPROVED** [review]
Looks very much better ! 

Surprised that we have to vendor in their components but 🤷 

Still think we should remove the use-client directive since we don't have RSC

**Comment by fontanierh** [comment]
> What do you mean by not a component library?
> I mean you do not install it as a dependency. It is not available or distributed via npm.
> Pick the components you need. Copy and paste the code into your project and customize to your needs. The code is yours.

That's pretty crazy

File: front/components/poke/shadcn/ui/data_table.tsx (hunk 1 above)
Comment (flvndvd) [reply front/components/poke/shadcn/ui/data_table.tsx:1]: Let me remove it.

**Pull request merged**